js/reptiles.js

```diff
@@ -0,0 +1,102 @@
+  
+  let animals = [frilledNeckLizard, hawksbillTurtle, perentie,];
+  
+  const getAllAnimals = () => {
+    animals.forEach(e => {
+      const animal_info = document.querySelector(".animal_names");
+      animal_info.innerHTML = animal_info.innerHTML + `<li>${e.name}</li>`;
+    });
+  }
+  
+  const onClick = {
+    lastClicked: null,
+    showWelcomeMsg: false,
+  }
+  
+  const showAnimal = (animalName) => {
+    const animalInfo = document.getElementById("content");
+    let animal = getAnimal(animalName);
+  
+  
+    if(onClick.lastClicked === animalName && !onClick.showWelcomeMsg){
+      animalInfo.innerHTML = `<h2>Embark on a Reptilian Adventure!</h2>
+      <p>Welcome, intrepid explorer! From the deserts to the rainforests, reptiles inhabit some of the most extreme and beautiful places on Earth. Let’s dive into their world and uncover the secrets of their survival.</p>`
+      onClick.showWelcomeMsg = true;
+    }else{
+    animalInfo.innerHTML = ` 
+      <img src="${animal.image}" alt="${animal.name}" class="animal_imge"> 
+      <h2 class="animal_name">${animal.name}</h2> 
+      <p>${animal.description.substring(0, 200)}
+      <span class="dots">...</span>
+      <span class="more hidden">${animal.description.substring(200)}</span></p>
+      <button class="read-more">Read more</button> 
+      <p><strong>Group:</strong> ${animal.group}</p>
+      <p><strong>Lifespan:</strong> ${animal.lifespan}</p>
+      <p><strong>Weight:</strong> ${animal.weight}</p>
+      <p><strong>Found:</strong> ${animal.found}</p>
+      <p><strong>Food:</strong> ${animal.food}</p>`;
+
+    // Add an event listener to the "Read more" button
+    const readMoreButton = animalInfo.querySelector(".read-more");
+    readMoreButton.addEventListener("click", function () {
+      const dots = animalInfo.querySelector(".dots");
+      const moreText = animalInfo.querySelector(".more");
+  
+      if (moreText.classList.contains("hidden")) {
+        // Show the hidden text and update button text
+        moreText.classList.remove("hidden");
+        dots.style.display = "none";
+        readMoreButton.textContent = "Read less";
+      } else {
+        // Hide the extra text and reset button text
+        moreText.classList.add("hidden");
+        dots.style.display = "inline";
+        readMoreButton.textContent = "Read more";
+      }
+    });
+  
+    onClick.lastClicked = animalName;
+    // Reset welcome message
+    onClick.showWelcomeMsg = false; 
+  }
+  }
+  
+  getAllAnimals();
+  
+  const items = document.querySelectorAll('li');
+  items.forEach(item => {
+    item.addEventListener('click', (e) => {
+      showAnimal(e.target.textContent);
+  
+    }
+    )
+  })
+  
+  
```

Comment:
remove some space

Reply:
I'm very careful abut spaces, may be it seems here. will double check